src/main/java/org/assertj/core/api/SoftAssertions.java

```diff
@@ -105,16 +106,43 @@
  */
 public class SoftAssertions extends AbstractStandardSoftAssertions {
 
-	/**
+  /**
    * Verifies that no proxied assertion methods have failed.
    *
    * @throws SoftAssertionError if any proxied assertion objects threw
    */
   public void assertAll() {
-	List<Throwable> errors = errorsCollected();
-	if (!errors.isEmpty()) {
-	  throw new SoftAssertionError(extractProperty("message", String.class).from(errors));
-	}
+    List<Throwable> errors = errorsCollected();
+    if (!errors.isEmpty()) {
+      throw new SoftAssertionError(createErrorMessagesWithLineNumbers(errors));
+    }
+  }
+
+  private ArrayList<String> createErrorMessagesWithLineNumbers(List<Throwable> errors) {
+    ArrayList<String> errorMessages = new ArrayList<>();
+    for (Throwable error : errors) {
+      removeReflectionAndCglibProxyElementsFromStackTrace(error);
+      errorMessages.add(createErrorMessageWithLineNumber(error));
+    }
+    return errorMessages;
   }
 
+  private String createErrorMessageWithLineNumber(Throwable error) {
+    StackTraceElement stackTraceElement = error.getStackTrace()[0];
+    return error.getMessage() + "at (" + stackTraceElement.getClassName() + ":"
+           + stackTraceElement.getLineNumber() + ")";
+  }
+
+  private void removeReflectionAndCglibProxyElementsFromStackTrace(Throwable throwable) {
+    List<StackTraceElement> filtered = newArrayList(throwable.getStackTrace());
+    for (StackTraceElement element : throwable.getStackTrace()) {
+      String className = element.getClassName();
+      if (className.startsWith("sun.reflect") || className.startsWith("java.lang.reflect")
+          || className.startsWith("net.sf.cglib.proxy")) {
```

Comment:
not sure the condition `className.startsWith("net.sf.cglib.proxy")` will ever be met as cglib is repackages inside assertj, use `className.startsWith("org.assertj.core.internal.cglib.proxy")` instead - you can verify that in assertj-examples.